File: GFG Sheet/Array/Palindromic Array.cpp
// { Driver Code Starts
#include<iostream>
#include<string.h>
using namespace std;
int PalinArray(int a[], int n);
int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;
		int a[n];
		for(int i = 0; i < n; i++)
			cin>>a[i];
		cout<<PalinArray(a,n)<<endl;
	}
}// } Driver Code Ends


/*Complete the function below*/
int PalinArray(int a[], int n)
{  //add code here.

for(int i=0;i<n;i++){
    auto s = std::to_string(a[i]);
    auto val=string(s.rbegin(),s.rend());
    if(val!=s)
    return false;
}
return true;
}

File: GFG Sheet/Array/Palindromic Array.py
# Your task is to complete this function //https://practice.geeksforgeeks.org/problems/palindromic-array-1587115620/1#
# Function should return True/False or 1/0
def PalinArray(a ,n):
    for x in range(n):
        val=str(a[x])
        if val!=val[::-1]:
            return False
        
    return True        
    # Code here




#{ 
#  Driver Code Starts
# Driver Program
if __name__=='__main__':
    t=int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        if PalinArray(arr, n):
            print(1)
        else:
            print(0)
# Contributed By: Harshit Sidhwa

# } Driver Code Ends
